src/core/qgsstacktrace.cpp

```diff
@@ -0,0 +1,115 @@
+/***************************************************************************
+  qgsstacktrace - %{Cpp:License:ClassName}
+
+ ---------------------
+ begin                : 30.3.2017
+ copyright            : (C) 2017 by Matthias Kuhn
+ email                : matthias@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsstacktrace.h"
+#include <stdlib.h>
+#include <execinfo.h>
+#include <cxxabi.h>
+
+#ifdef QGISDEBUG
+#ifdef Q_OS_LINUX
+QStringList QgsStacktrace::trace( unsigned int maxFrames )
```

Comment:
There's already code in [src/app/main.cpp:196](https://github.com/qgis/QGIS/blob/master/src/app/main.cpp#L196) to show backtraces (including windows).